lib/api/core/validation/index.js

```diff
@@ -1,690 +1,682 @@
+      );
+    }
 
-/**
- * @param {string} fieldName
- * @param {*} documentSubset
- * @param {StructuredFieldSpecification} collectionSpecSubset
- * @param {boolean} strictness
- * @param {string[]} errorMessages
- * @param {boolean} verbose
- * @returns {boolean}
- */
-Validation.prototype.isValidField = function validationIsValidField (fieldName, documentSubset, collectionSpecSubset, strictness, errorMessages, verbose) {
-  var
-    nestedStrictness,
-    /** @type StructuredFieldSpecification */
-    field = collectionSpecSubset[fieldName],
-    fieldValues,
-    i,
-    fieldErrors,
-    result = true;
-
-  if (field.mandatory && !field.hasOwnProperty('defaultValue') && (!documentSubset.hasOwnProperty(fieldName) || typeof documentSubset[fieldName] === 'undefined' || documentSubset[fieldName] === null)) {
-    manageErrorMessage(field.path, errorMessages, 'The field is mandatory.', verbose);
-    return false;
+    // We try to validate every field in order to get all error messages if any
+    return Object.keys(collectionSpecSubset).reduce((reductionResult, fieldName) =>
+      this.isValidField(fieldName, documentSubset, collectionSpecSubset, strictness, errorMessages, verbose) && reductionResult
+    , true);
   }
 
-  if (documentSubset.hasOwnProperty(fieldName) && typeof documentSubset[fieldName] !== 'undefined' && documentSubset[fieldName] !== null) {
-    if (field.multivalued.value) {
-      if (!Array.isArray(documentSubset[fieldName])) {
-        manageErrorMessage(field.path, errorMessages, 'The field must be multivalued, unary value provided.', verbose);
-        return false;
-      }
-
-      if (field.multivalued.hasOwnProperty('minCount') && documentSubset[fieldName].length < field.multivalued.minCount) {
-        manageErrorMessage(field.path, errorMessages, `Not enough elements. Minimum count is set to ${field.multivalued.minCount}.`, verbose);
-        return false;
-      }
-
-      if (field.multivalued.hasOwnProperty('maxCount') && documentSubset[fieldName].length > field.multivalued.maxCount) {
-        manageErrorMessage(field.path, errorMessages, `Too many elements. Maximum count is set to ${field.multivalued.maxCount}.`, verbose);
-        return false;
-      }
+  /**
+   * @param {string} fieldName
+   * @param {*} documentSubset
+   * @param {StructuredFieldSpecification} collectionSpecSubset
+   * @param {boolean} strictness
+   * @param {string[]} errorMessages
+   * @param {boolean} verbose
+   * @returns {boolean}
+   */
+  isValidField(fieldName, documentSubset, collectionSpecSubset, strictness, errorMessages, verbose) {
+    /** @type StructuredFieldSpecification */
+    const field = collectionSpecSubset[fieldName];
+    let result = true;
 
-      fieldValues = documentSubset[fieldName];
+    if (field.mandatory && !field.hasOwnProperty('defaultValue') && (!documentSubset.hasOwnProperty(fieldName) || typeof documentSubset[fieldName] === 'undefined' || documentSubset[fieldName] === null)) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1067)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 204 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)